docs/modules/ROOT/pages/gsnkit.adoc

```diff
@@ -23,75 +15,97 @@ starter kit you want to use. In this case we'll start our project using the kit
-npm run build
-``
+## Why this kit?
+
+This kit leverages GSN to create dapps that are ready for mass adoption. Free your users from
+the initial burden of installing Metamask and obtaining Ether. Create blockchain applications
+that are indistinguishable from Web2.0 apps.
+
+This documents assumes familiarity with Gas Station Network. Here are some resources about it:
+
+https://gsn.openzeppelin.com/[Website]
+https://docs.openzeppelin.com/contracts/2.x/gsn[GSN Contracts Overview]
+https://github.com/OpenZeppelin/openzeppelin-gsn-provider[GSN Provider]
+
+### How does it use Web3 with GSN?
+This kit uses Open Zeppelin https://github.com/OpenZeppelin/openzeppelin-network.js[network.js] to create the connection to Web3. Using a couple
+flags for development and production you can see how the dapp obtains a context that is aware of Gas Station Network.
+
+[source,solidity]
+----
+// get GSN web3
+const context = useWeb3Network('http://127.0.0.1:8545', {
+  gsn: {
+    dev: true,
+  },
+});
+----
+
+### How are the contracts modified to use GSN?
+
+The `Counter` contract is modified to inherit from `RelayRecipient`.
+Also, the counter contract is going to naively pay for all the transactions that are submitted.
+Note how the `acceptRelayedCall` determines this by returning 0.
+
+[source,solidity]
+----
+pragma solidity ^0.5.0;
+
+import "tabookey-gasless/contracts/GsnUtils.sol";
+import "tabookey-gasless/contracts/IRelayHub.sol";
+import "tabookey-gasless/contracts/RelayRecipient.sol";
+import "@openzeppelin/upgrades/contracts/Initializable.sol";
+
+contract Counter is Initializable, RelayRecipient {
+  //it keeps a count to demonstrate stage changes
+  uint private count;
+  address private _owner;
+
+  function initialize(uint num) public initializer {
+    _owner = msg.sender;
+    count = num;
+  }
+
+  function getHubAddr() public view returns (address) {
+    return address(0xD216153c06E857cD7f72665E0aF1d7D82172F494);
+  }
+
+  // accept all requests
+  function acceptRelayedCall(
+    address,
+    address,
+    bytes calldata,
+    uint256,
+    uint256,
+    uint256,
+    uint256,
+    bytes calldata,
+    uint256
+    ) external view returns (uint256, bytes memory) {
+    return (0, "free");
+  }
+  ...
+}
+----
+
+### How to know if my recipient has funds?
+
+The frontend also has some functions to help you see how much remaining balance you have left.
+Once it runs out, transactions will stop working because your dapp won't be able to pay the gas fee
+on behalf of its users.
+
+[source,solidity]
+----
+const getDeploymentAndFunds = async () => {
+  if (instance) {
+    const isDeployed = await isRelayHubDeployedForRecipient(lib, _address);
+    setIsDeployed(isDeployed);
+    if (isDeployed) {
+      const funds = await getRecipientFunds(lib, _address);
+      setFunds(funds);
+    }
+  }
+};
+----
+
+You can top your balance by sending funds to your contract using `oz transfer` command.
```

Comment:
Watch out! A transfer doesn't cut it. You actually need to call a specific method on the hub itself. You can use `npx oz-gsn fund-recipient --recipient ADDRESS`, or go to the dapp tool to do it.

Reply:
changed